src/atom_finder/classifier/post-increment-decrement.clj

```diff
@@ -0,0 +1,34 @@
+(in-ns 'atom-finder.classifier)
+(import '(org.eclipse.cdt.core.dom.ast IASTNode IASTUnaryExpression
+           IASTForStatement IASTExpressionStatement IASTConditionalExpression))
+
+
+(defn post-increment-decrement?
+  "Is this oprator post decrement or increment?"
+  [node]
+  (let [u-ops #{ ; postfix decrement or increment operators
+                IASTUnaryExpression/op_postFixDecr       IASTUnaryExpression/op_postFixIncr}]
+    (and (instance? IASTUnaryExpression node) (contains? u-ops (.getOperator node)))))
+  
+
+(defn post-increment-decrement-atom?
+  "Does this AST node an statement that contains decrement and increment operators?"
+  [node]
+  (let [statement #{;statement types that might not cause confusion
+                    IASTForStatement IASTExpressionStatement IASTConditionalExpression}]
+      (cond
+        (leaf? node) false
+        
+        (not-any? #(instance? % node) statement)
+        (if(some true? (map post-increment-decrement? (children node))) true false)
+        
+        :else false)))
```

Comment:
The functions in this file seem to be copy-pasted a lot, can you factor out the common logic into shared functions?